release-23.2: sql/logictest: deflake schema_change_in_txn

rafiss opened the pull request:
Backport 1/1 commits from #129013 and #129487 on behalf of @mgartner.

/cc @cockroachdb/release

----

In the update test, there's a chance that a job succeeds quickly, so we
account for that by matching statuses of "running" or "succeeded".

Fixes #128746

Release note: None

----

This is a follow-up to https://github.com/cockroachdb/cockroach/pull/129013 which changes the incorrectly used
similarity operator, `%`, to the regex match operator, `~`.

Release note: None

----

Release justification: test only change

mgartner committed:
sql/logictest: deflake schema_change_in_txn

In the update test, there's a chance that a job succeeds quickly, so we
account for that by matching statuses of "running" or "succeeded".

Fixes #128746

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129432)
<!-- Reviewable:end -->

mgartner committed:
sql/logictest: fix incorrect operator in test

This is a follow-up to #129013 which changes the incorrectly used
similarity operator, `%`, to the regex match operator, `~`.

Release note: None
